framework/doc/content/source/bcs/ADFunctionDirichletBC.md

```diff
@@ -30,6 +30,12 @@ corresponding to the boundary subset $\partial \Omega_D$ via the
 
 Note that `ADFunctionDirichletBC` computes its Jacobian using automatic differentiation.
 
+## Preset boundary conditions
+
+With the parameter `preset = true`, the value of the boundary condition is applied
+before the solve begins. With `preset = false` (default), the boundary condition is
+only enforced as the solve progresses. In certain situations, one is better than another.
+
```

Comment:
I think this doco should be expanded a little.  The sentence "In certain situations, one is better than another" isn't super useful to users.  I suspect anyone reading this will just try `preset = true` and `preset = false` and see which happens to be best for them.  Perhaps if you included one example of where "true" is best, and another example of where "false" is best, explaining why, it'd help users more.

Reply:
I spoke to @friedmud some about this. I'm not incredibly familiar with the actual _use_ of presetting the BCs. The conversation led to the thought that they're great in most situations and that it would work to only show an example where it doesn't work well: in a solid mechanics problem where you can actually tangle your mesh with a preset BC. What are your thoughts? Can you think of other examples?

Reply:
My experience is that preset BCs are generally better. In some highly nonlinear formulations, i.e. finite deformation + contact, even though the global minimizer is unique, there are many local minimizers in the vicinity, and numerically the solve may find a local minimizer which satisfies given tolerances but is actually a worse solution than the global minimizer. For example, if you take a very large step, the preset BC may stretch too much such that you have an overly distorted mesh, and nonlinear iterations become difficult from there.

Reply:
I concur with @hugary1995 .  My experience is that presetBC is almost always better, and I will almost always use `preset = true` (i don't know why you chose `false` to be the default).  Certainly it's better if you're doing something like `value=0` at the start of a simulation, where the initial condition is zero.  However, there are situations where presetBC forces a timestep to initialize in a way-out-of-equilibrium configuration (which can especially happen with things like `FunctionPresetBC`) and MOOSE finds it difficult to converge.